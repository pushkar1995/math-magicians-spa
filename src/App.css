.app {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.calculator-container {
  border: 1px solid black;
  width: 30vw;
  height: 40vh;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.screen {
  background-color: grey;
  color: rgb(225, 225, 225);
  padding: 0.7rem;
  height: 2.5rem;
  width: 100%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  font-size: 1.2rem;
  box-sizing: border-box;
}

.buttonContainer {
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

button {
  border: 1px solid grey;
  background-color: rgb(212, 212, 212);
  color: rgb(39, 60, 60);
}

.zero {
  grid-column-start: 1;
  grid-column-end: 3;
}

.button-container > *:nth-child(4n+4),
.equal {
  background-color: orange;
}
