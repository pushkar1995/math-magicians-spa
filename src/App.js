import './App.css';
import Calculator from './components/calculator.js';

function App() {
  return (
    <div className="app">
      <Calculator />
    </div>
  );
}

export default App;