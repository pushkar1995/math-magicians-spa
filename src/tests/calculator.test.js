import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Calculator from '../components/calculator';
import App from '../App';


const customTextMatcher = (content, element) => {
  // Implement your custom matching logic here
  return element.textContent === content;
};


describe('Renders correctly', () => {
  test('Test CalculatorComponent', () => {
    const tree = render(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('Renders correctly', () => {
  test('User clicks on the number 3 on the calculator', () => {
    const { container } = render(<App />);
    userEvent.click(screen.getAllByText('Calculator')[0]);
    // console.log(userEvent);
    userEvent.click(screen.getByText("3"));
    // console.log(userEvent);
    expect(container.getElementsByClassName('screen')[0].firstChild.innerHTML).toContain("3");
  });

  test('User clicks on the number 5 and then clicks AC', () => {
    const { container } = render(<App />);
    userEvent.click(screen.getAllByText('Calculator')[0]);
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('AC'));
    expect(container.getElementsByClassName('screen')[0].firstChild.innerHTML).toBe('  ');
  });

  // test('User clicks on the number 5 and then +/-', () => {
  //   const { container } = render(<App />);
  //   userEvent.click(screen.getAllByText('Calculator')[0]);
  //   userEvent.click(screen.getByText('5'));
  //   expect(container.getElementsByClassName('screen')[0].firstChild.innerHTML).toContain('5');
  //   userEvent.click(screen.getByText('+/-'));
  //   expect(container.getElementsByClassName('screen')[0].firstChild.innerHTML).toContain('-5');
  // });

});
