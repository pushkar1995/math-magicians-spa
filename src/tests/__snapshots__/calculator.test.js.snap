// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly Test CalculatorComponent 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator-container"
      >
        <div
          class="screen"
        >
          <span>
             
             
          </span>
        </div>
        <div
          class="buttonContainer"
        >
          <button
            class="operator"
            type="button"
          >
            AC
          </button>
          <button
            class="operator"
            type="button"
          >
            +/-
          </button>
          <button
            class="operator"
            type="button"
          >
            %
          </button>
          <button
            class="operator"
            type="button"
          >
            รท
          </button>
          <button
            class="number"
            type="button"
          >
            7
          </button>
          <button
            class="number"
            type="button"
          >
            8
          </button>
          <button
            class="number"
            type="button"
          >
            9
          </button>
          <button
            class="operator"
            type="button"
          >
            x
          </button>
          <button
            class="number"
            type="button"
          >
            4
          </button>
          <button
            class="number"
            type="button"
          >
            5
          </button>
          <button
            class="number"
            type="button"
          >
            6
          </button>
          <button
            class="operator"
            type="button"
          >
            -
          </button>
          <button
            class="number"
            type="button"
          >
            1
          </button>
          <button
            class="number"
            type="button"
          >
            2
          </button>
          <button
            class="number"
            type="button"
          >
            3
          </button>
          <button
            class="operator"
            type="button"
          >
            +
          </button>
          <button
            class="number zero"
            type="button"
          >
            0
          </button>
          <button
            class="number"
            type="button"
          >
            .
          </button>
          <button
            class="operator equal"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator-container"
    >
      <div
        class="screen"
      >
        <span>
           
           
        </span>
      </div>
      <div
        class="buttonContainer"
      >
        <button
          class="operator"
          type="button"
        >
          AC
        </button>
        <button
          class="operator"
          type="button"
        >
          +/-
        </button>
        <button
          class="operator"
          type="button"
        >
          %
        </button>
        <button
          class="operator"
          type="button"
        >
          รท
        </button>
        <button
          class="number"
          type="button"
        >
          7
        </button>
        <button
          class="number"
          type="button"
        >
          8
        </button>
        <button
          class="number"
          type="button"
        >
          9
        </button>
        <button
          class="operator"
          type="button"
        >
          x
        </button>
        <button
          class="number"
          type="button"
        >
          4
        </button>
        <button
          class="number"
          type="button"
        >
          5
        </button>
        <button
          class="number"
          type="button"
        >
          6
        </button>
        <button
          class="operator"
          type="button"
        >
          -
        </button>
        <button
          class="number"
          type="button"
        >
          1
        </button>
        <button
          class="number"
          type="button"
        >
          2
        </button>
        <button
          class="number"
          type="button"
        >
          3
        </button>
        <button
          class="operator"
          type="button"
        >
          +
        </button>
        <button
          class="number zero"
          type="button"
        >
          0
        </button>
        <button
          class="number"
          type="button"
        >
          .
        </button>
        <button
          class="operator equal"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
